### PR TITLE
go-errcheck improvements

### PR DESCRIPTION
**Pull request by dominikh**
Don't treat errors as warnings and simplify a pattern.

I'm not sure if the new test is correct. I can't get tests to run locally at the moment.

### CODE REVIEW

File: test/flycheck-test.el [hunk 1]
```diff
@@ -3471,6 +3471,11 @@ See https://github.com/flycheck/flycheck/issues/531 and Emacs bug #19206"))
      '(7 9 warning "Ignored `error` returned from `f.Close()`"
          :checker go-errcheck)
      '(9 9 warning "Ignored `error` returned from `os.Stat(\"enoent\")`"
+         :checker go-errcheck))
+
+    (flycheck-ert-should-syntax-check
```
Comment (lunaryorn) [inline test/flycheck-test.el]: Please move this into a new test case.

File: flycheck.el [hunk 2]
```diff
@@ -6733,18 +6733,23 @@ or newer.
...
             line-end))
   :error-filter
   (lambda (errors)
     (let ((errors (flycheck-sanitize-errors errors)))
       (dolist (err errors)
-        (-when-let (message (flycheck-error-message err))
-          ;; Improve the messages reported by errcheck to make them more clear.
-          (setf (flycheck-error-message err)
-                (format "Ignored `error` returned from `%s`" message)))))
+        (if (eq (flycheck-error-level err) 'warning)
```
Comment (lunaryorn) [inline flycheck.el]: Please use `when` instead of `if`s with no "else" branch.

**Comment by lunaryorn** [comment]
@dominikh Thank you very much.  I've left a couple of remarks :)

Mind if I ask why you can't get the Go tests to run?  Is there anything I can help with?

**Comment by dominikh** [comment]
> Mind if I ask why you can't get the Go tests to run? Is there anything I can help with?

I can't get the test suite to run at all and haven't yet had the time to investigate why. Here's the output I get:

```
$ make specs                                                                               
cask exec emacs -Q --batch -L .  -f batch-byte-compile flycheck.el
Warning: Lisp directory `/usr/share/emacs/24.5/lisp/emacs-parallel': No such file or directory
Warning: Lisp directory `/usr/share/emacs/24.5/lisp/emacs-parallel': No such file or directory

In toplevel form:
flycheck.el:74:13:Error: Cannot open load file: no such file or directory, let-alist
Makefile:62: recipe for target 'flycheck.elc' failed
make: *** [flycheck.elc] Error 1
```

**Comment by dominikh** [comment]
@lunaryorn I have addressed your comments. I've already squashed the changes because they were minor ones.

**Comment by lunaryorn** [comment]
@dominikh Did you run `make init` before?

**Comment by dominikh** [comment]
@lunaryorn Okay, that fixes that issue, thanks. Is it expected that `make specs` requires user input when it tries to decrypt a GPG file? It's prompting me for a password. Upon entering garbage, a test fails:

```
Utilities
  flycheck-encrypted-buffer-p
    considers a temporary buffer as unencrypted
    considers a file buffer as unencrypted
    recognizes an encrypted bufferDecrypting /tmp/flycheck-encrypted-file22761G4L.txt.gpg...0%
Decrypting /tmp/flycheck-encrypted-file22761G4L.txt.gpg...done
  FAILED
```

Maybe we should move this discussion to a separate issue or forum.

**Comment by lunaryorn** [comment]
@dominikh It shouldn't, and I tried to prevent that.  That's a matter for a separate issue, through.

For this PR it should be sufficient to check `make LANGUAGE=go integ`, i.e. the integration tests for Go.

File: test/flycheck-test.el [hunk 3]
```diff
@@ -3473,6 +3473,14 @@ See https://github.com/flycheck/flycheck/issues/531 and Emacs bug #19206"))
      '(9 9 warning "Ignored `error` returned from `os.Stat(\"enoent\")`"
          :checker go-errcheck))))
 
+(flycheck-ert-def-checker-test go-errcheck go nil
```
Comment (lunaryorn) [inline test/flycheck-test.el]: `nil` must be a proper test name here, e.g. "error" or so.

Comment (dominikh) [inline test/flycheck-test.el]: I simply copied from the existing errcheck test. Does it only require names when there's more than one test for a checker? Or is this a stylistic issue?

Comment (lunaryorn) [inline test/flycheck-test.el]: @dominikh The name must be different if there are to be multiple tests for the same checker.  `nil` simply means "default" here.

Comment (lunaryorn) [inline test/flycheck-test.el]: @dominikh Did you get a chance to update the test name already?

Comment (dominikh) [inline test/flycheck-test.el]: Sorry, I've been rather busy this weekend. I'll get to it in the next 2 days and ping you when I've done so.

Comment (lunaryorn) [inline test/flycheck-test.el]: @dominikh Great, thanks :blush:

Comment (lunaryorn) [inline test/flycheck-test.el]: @dominikh Sorry to ping you, but any news here?

**Comment by lunaryorn** [comment]
@dominikh What's the status of this?

**Comment by lunaryorn** [comment]
@dominikh I'll close this pull request as there's no activity here anymore.  I'll happily re-open it if you pick it up again.

Sorry.

**Pull request closed without merging**